internal/service/secretsmanager/secret.go

```diff
@@ -269,6 +270,11 @@ func resourceSecretRead(ctx context.Context, d *schema.ResourceData, meta any) d
 	})
 
 	if err != nil {
+		if strings.Contains(err.Error(), "contains invalid principals") {
+			d.Set(names.AttrPolicy, "")
+			return sdkdiag.AppendWarningf(diags, "reading Secrets Manager Secret (%s) policy: %s", d.Id(), err)
```

Comment:
Should this append the warning without returning so that `setTagsOut` can still be run?

Reply:
@jar-b updated to return only if there is a diagnostic error